File: 768-max-chunks-to-make-sorted-ii/768-max-chunks-to-make-sorted-ii.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        s1, s2, res = 0, 0, 0
        for a, b in zip(arr, sorted(arr)):
            s1 += a
            s2 += b
            res += s1 == s2
        return res
        

File: 768-max-chunks-to-make-sorted-ii/768-max-chunks-to-make-sorted-ii.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maxChunksToSorted = function(arr) {
  const sorted = [...arr].sort((a, b) => a - b);
  let [sum1, sum2, res] = [0, 0, 0];
  
  for (let i = 0; i < arr.length; i++) {
    sum1 += arr[i];
    sum2 += sorted[i];
    
    res += sum1 === sum2;
  }
  
  return res;
};